### DISCUSSION TITLE
Add ability to prevent SVG assets from being transformed to raster image formats

### DISCUSSION THREAD

**Discussion by Wiejeben** [craftcms/cms#12219] [Ideas] [answered]
Hi!

In certain scenarios I would like to have the ability to always be provided with the (source) SVG file even if a WebP, JPG, PNG, etc transform is being applied. This could drastically simplify retaining the often lower file size vector based images.

Currently because GraphQL doesn't have conditions we are always transforming SVG's into WebP's, in order to still access the source SVG file we solve this the following way:
```gql
labelLogoOriginal: labelLogo {
    extension
    title
    url
}
labelLogo @transform(handle:"labelImage") {
    title
    url
    focalPoint
    srcset(sizes: ["2x"])
}
```

So when the `labelLogoOriginal.extension` is 'svg' we would use `labelLogoOriginal` for our `<img>`, otherwise `labelLogo` would be used instead. How this would work in combination with srcset I don't know, I'm guessing it would just provide the same SVG file multiple times. Or maybe the different srcset sizes could be eliminated entirely.

Regardless of GraphQL of Twig it always adds additional complexity to something that I would prefer to not happen in the first place anyway.

Because of this I would suggest adding an additional parameter to the asset transform saying something like "Prevent transforming vector based images", preventing the transformation entirely.

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
Sounds like the [`transformSvgs`](https://craftcms.com/docs/4.x/config/general.html#transformsvgs) setting will do the trick?

**Reply by Wiejeben** [reply]
Oh yeah that totally does, somehow didn’t catch into that change. Thanks!
